### PR TITLE
Initial take at new documentation for path factoring

### PR DESCRIPTION
**Pull request by msullivan**
Please give feedback.

### CODE REVIEW

File: docs/edgeql/path_resolution.rst [hunk 1]
```diff
@@ -1,8 +1,183 @@
...
+However, when a path is used multiple times in "sibling" contexts,
+a cross-product will be computed:
+
+.. code-block:: edgeql-repl
+
+    db> select User.first_name ++ ' ' ++ User.last_name;
+    {'Peter Parker', 'Peter Stark', 'Tony Parker', 'Tony Stark'}
+
+
+If the old behavior is desired, you can typically rewrite the query
```
Comment (dnwpark) [inline docs/edgeql/path_resolution.rst:58]: Maybe phrase this as, `If the "sibling" contexts should be bound together, you can typically ..... This was the old behaviour.`

File: docs/edgeql/path_resolution.rst [hunk 2]
```diff
@@ -1,8 +1,183 @@
...
+
+Discussion of rationale for this change is available in
+`the RFC <rfc_>`_.
+
+
+New path scoping
+----------------
+
+When applying a shape to a path (or to a path that has shapes applied
+to it already), the path will be be bound inside computed
```
Comment (dnwpark) [inline docs/edgeql/path_resolution.rst:25]: Maybe needs a more precise definition of `bound`.

**Review by aljazerzen — APPROVED** [review]
This reads well for me, but I'm not confident that average edgedb user will understand the change in behavior. That said, they don't need to - the upgrade plan is the only really important for them.

Maybe add a TLDR at the top?

File: docs/edgeql/path_resolution.rst [hunk 3]
```diff
@@ -1,8 +1,183 @@
...
+.. code-block:: edgeql-repl
+
+    db> select User {
+    ...   name := User.first_name ++ ' ' ++ User.last_name
+    ... }
+    {User {name: 'Peter Parker'}, User {name: 'Tony Stark'}}
+
+
+When doing ``SELECT``, ``UPDATE``, or ``DELETE``, if the subject is a
+path with zero or more shapes applied to it, the path will be
```
[the remark this answers is not in the document]
Comment (msullivan) [reply docs/edgeql/path_resolution.rst:37]: Yeah, zero or more!

Maybe instead I should say "is a path (or a path with shapes applied to it)"?
Or "is a path, optionally with shapes applied to it,"?

**Pull request merged**